packages/fxa-payments-server/src/components/PlanErrorDialog/index.test.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { render, cleanup } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
+
+import PlanErrorDialog from './index';
+import { PLANS } from '../../lib/mock-data';
+
+const locationReload = () => {};
+const plans = {
+  error: null,
+  loading: false,
+  result: null,
+};
+
+afterEach(cleanup);
+
+describe('PlanErrorDialog', () => {
+  it('renders as expected for no plan for product', () => {
+    const subject = () => {
+      return render(
+        <PlanErrorDialog
+          {...{
+            locationReload,
+            plans,
+          }}
+        />
+      );
+    };
+    const { queryByTestId } = subject();
+    const errorMessage = queryByTestId('error-loading-plans');
+    expect(errorMessage).toBeInTheDocument();
+  });
+
+  it('renders as expected for no selectedPlan', () => {
+    const subject = () => {
+      return render(
+        // @ts-ignore
+        <PlanErrorDialog
+          {...{
+            locationReload,
+            plans: { ...plans, result: PLANS },
```

Comment:
I get a mystifying TS error if I don't ignore, even though AFAICT this `plans` object matches the `FetchStateLoaded` type:
```
Type '{ result: Plan[]; error: null; loading: boolean; }' is not assignable to type 'FetchStateLoaded<Plan[]>'.
        Types of property 'loading' are incompatible.
        Type 'boolean' is not assignable to type 'false'.ts(2322)
```

Reply:
I figured out a solution:
```diff
diff --git a/packages/fxa-payments-server/src/components/PlanErrorDialog/index.test.tsx b/packages/fxa-payments-server/src/components/PlanErrorDialog/index.test.tsx
index 1d3c6bfe0..ed020d167 100644
--- a/packages/fxa-payments-server/src/components/PlanErrorDialog/index.test.tsx
+++ b/packages/fxa-payments-server/src/components/PlanErrorDialog/index.test.tsx
@@ -4,9 +4,10 @@ import '@testing-library/jest-dom/extend-expect';
 
 import PlanErrorDialog from './index';
 import { PLANS } from '../../lib/mock-data';
+import { FetchState, Plan } from '../../store/types';
 
 const locationReload = () => {};
-const plans = {
+const plans: FetchState<Plan[], any> = {
   error: null,
   loading: false,
   result: null,
@@ -34,7 +35,6 @@ describe('PlanErrorDialog', () => {
   it('renders as expected for no selectedPlan', () => {
     const subject = () => {
       return render(
-        // @ts-ignore
         <PlanErrorDialog
           {...{
             locationReload,
```

Reply:
@biancadanforth wonderful, thank you :)